libguard/guard_interface.cpp

```diff
@@ -139,19 +143,33 @@ static GuardRecord getHostEndiannessRecord(const GuardRecord& record)
 static void fileSizeCheck(size_t size, uint32_t count)
 {
     GuardRecord record;
+    int pos = 0;
+    int lastPos = 0;
     uint32_t aval_size = size - (count * sizeof(record) + headerSize);
 
     if (aval_size < sizeof(record))
     {
-        guard_log(GUARD_ERROR, "Size left in GUARD file is %d\n", aval_size);
-        throw GuardFileOverFlowed("Enough size is not available in GUARD file");
+        // Delete the oldest resolved guard record, so that new guard
```

Comment:
I would prefer this can be handle in create API rather than doing here - IMO we are losing the function scope of "fileSizeCheck"

JS - This can be a separate commit to revert if not required - Your wish

Reply:
added changes in create api